Minesweeper.py

```diff
@@ -175,10 +180,29 @@ def __init__(self):
         self.gamestate = GameState()
         self.tiles = Tiles(self)
 
+    def handle_game_events(self, events):
+        if self.gamestate.tutorial:
+            self.tutorial_screen.handle_events(events)
+
+        if self.gamestate.game or self.gamestate.game_over or self.gamestate.game_won:
+            mouse_button_events = [event for event in events if event.type == pygame.MOUSEBUTTONUP]
+            for event in mouse_button_events:
+                if event.button == 1:
+                    mouse_pos = pygame.mouse.get_pos()
+                    if not self.gamestate.game_over and self.tiles.collidepoint(mouse_pos):
+                        if self.start_time == 0:
+                            self.start_time = pygame.time.get_ticks()
+                        self.tiles.uncover(mouse_pos)
+                    self.reset_button.handle_click(mouse_pos)
+                elif event.button == 3:
+                    mouse_pos = pygame.mouse.get_pos()
+                    if not self.gamestate.game_over and self.tiles.collidepoint(mouse_pos):
+                        self.tiles.flag(mouse_pos)
```

Comment:
same here, it would be better to use a helper function with a good name so that the intent of this code block is clear.